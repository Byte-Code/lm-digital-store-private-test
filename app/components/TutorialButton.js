import React, { Component, PropTypes } from 'react';
import Dialog from 'material-ui/Dialog';
import Slick from 'react-slick';
import styled from 'styled-components';

const Slide = styled.div`
  width: 1000px;
  height: 1420px;
  display: flex;
  align-items: center;
  justify-content: center;
  margin-right: 40px;
`;
const Img = styled.img`
  display: block;
  width: 1000px;
  height: 1420px;
`;

class TutorialButton extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired
  }

  state = {
    open: false,
  }

  componentDidUpdate() {
    window.requestAnimationFrame(() => {
      window.dispatchEvent(new Event('resize'));
    });
  }

  getSlickSettings = () => ({
    centerMode: true,
    arrows: false,
    infinite: false,
    variableWidth: true,
    dots: false,
    initialSlide: 0,
  })

  closeDialog = () => {
    this.setState({
      open: false
    });
  }

  openDialog = () => {
    this.setState({
      open: true
    });
  }

  render() {
    const settings = this.getSlickSettings();
    return (
      <div onTouchTap={this.openDialog}>
        {this.props.children}
        <Dialog
          modal={false}
          onRequestClose={this.closeDialog}
          open={this.state.open}
          contentClassName="similarDialog"
          contentStyle={{ width: '100%', maxWidth: 'none', background: 'transparent' }}
          bodyStyle={{ padding: 0, background: 'transparent' }}
        >
          <Slick {...settings}>
            <Slide><Img src={require('../assets/tutorial/tutorial1.png')} alt="tutorial 1" /></Slide>
            <Slide><Img src={require('../assets/tutorial/tutorial2.png')} alt="tutorial 2" /></Slide>
            <Slide><Img src={require('../assets/tutorial/tutorial3.png')} alt="tutorial 3" /></Slide>
            <Slide><Img src={require('../assets/tutorial/tutorial4.png')} alt="tutorial 4" /></Slide>
          </Slick>
        </Dialog>

      </div>
    );
  }
}

export default TutorialButton;
