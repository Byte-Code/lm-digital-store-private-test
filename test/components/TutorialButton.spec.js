import React from 'react';
import { shallow } from 'enzyme';

import TutorialButton, { Wrapper } from '../../app/components/TutorialButton';

const children = <button>click me</button>;

describe('TutorialButton', () => {
  it('should initially render with a closed dialog', () => {
    const result = shallow(
      <TutorialButton>
        {children}
      </TutorialButton>
    );
    expect(result.state('open')).toBe(false);
    expect(result).toMatchSnapshot();
  });

  it('should open the dialog once clicked', () => {
    const result = shallow(
      <TutorialButton>
        {children}
      </TutorialButton>
    );
    result.find(Wrapper).simulate('click');
    expect(result.state('open')).toBe(true);
    expect(result).toMatchSnapshot();
  });
});
