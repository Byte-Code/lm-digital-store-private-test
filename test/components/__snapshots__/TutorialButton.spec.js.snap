// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TutorialButton should initially render with a closed dialog 1`] = `
<styled.div
  onClick={[Function]}
>
  <button>
    click me
  </button>
  <Dialog
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    bodyStyle={
      Object {
        "background": "transparent",
        "padding": 0,
      }
    }
    contentClassName="similarDialog"
    contentStyle={
      Object {
        "background": "transparent",
        "height": 1420,
        "maxWidth": "none",
        "width": "100%",
      }
    }
    modal={false}
    onRequestClose={[Function]}
    open={false}
    repositionOnUpdate={true}
  >
    <CloseButton
      backgroundColor="#333333"
      fill="#858585"
      handleClick={[Function]}
      right={0}
      top={-250}
    />
    <Slider
      arrows={false}
      centerMode={true}
      dots={false}
      infinite={false}
      initialSlide={0}
      variableWidth={true}
    >
      <div>
        <styled.div>
          <styled.img
            alt="tutorial 1"
            src="test-file-stub"
          />
        </styled.div>
      </div>
      <div>
        <styled.div>
          <styled.img
            alt="tutorial 2"
            src="test-file-stub"
          />
        </styled.div>
      </div>
      <div>
        <styled.div>
          <styled.img
            alt="tutorial 3"
            src="test-file-stub"
          />
        </styled.div>
      </div>
      <div>
        <styled.div>
          <styled.img
            alt="tutorial 4"
            src="test-file-stub"
          />
        </styled.div>
      </div>
    </Slider>
  </Dialog>
</styled.div>
`;

exports[`TutorialButton should open the dialog once clicked 1`] = `
<styled.div
  onClick={[Function]}
>
  <button>
    click me
  </button>
  <Dialog
    autoDetectWindowHeight={true}
    autoScrollBodyContent={false}
    bodyStyle={
      Object {
        "background": "transparent",
        "padding": 0,
      }
    }
    contentClassName="similarDialog"
    contentStyle={
      Object {
        "background": "transparent",
        "height": 1420,
        "maxWidth": "none",
        "width": "100%",
      }
    }
    modal={false}
    onRequestClose={[Function]}
    open={true}
    repositionOnUpdate={true}
  >
    <CloseButton
      backgroundColor="#333333"
      fill="#858585"
      handleClick={[Function]}
      right={0}
      top={-250}
    />
    <Slider
      arrows={false}
      centerMode={true}
      dots={false}
      infinite={false}
      initialSlide={0}
      variableWidth={true}
    >
      <div>
        <styled.div>
          <styled.img
            alt="tutorial 1"
            src="test-file-stub"
          />
        </styled.div>
      </div>
      <div>
        <styled.div>
          <styled.img
            alt="tutorial 2"
            src="test-file-stub"
          />
        </styled.div>
      </div>
      <div>
        <styled.div>
          <styled.img
            alt="tutorial 3"
            src="test-file-stub"
          />
        </styled.div>
      </div>
      <div>
        <styled.div>
          <styled.img
            alt="tutorial 4"
            src="test-file-stub"
          />
        </styled.div>
      </div>
    </Slider>
  </Dialog>
</styled.div>
`;
